5-design-patterns/shipment/letter.ts

```diff
@@ -0,0 +1,26 @@
+import { IIdGenerator } from '../persistence/id-generator.model';
+import { Shipper } from '../shipper/Shipper';
+import { Shipment } from './Shipment';
+import { IShipment, IShipmentDTO, ShipmentType } from './shipment.model';
+
+export class Letter extends Shipment implements IShipment {
```

Comment:
```suggestion
export class Letter extends Shipment {
```
This class already extends `Shipment`, you don't need to implement `IShipment`

Reply:
Done.